src/Notes/WC_Admin_Notes_Test_Checkout.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * WooCommerce Admin Test Checkout.
+ *
+ * Adds a note to remind the user to test their store checkout.
+ *
+ * @package WooCommerce Admin
+ */
+
+namespace Automattic\WooCommerce\Admin\Notes;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Admin_Notes_Test_Checkout
+ */
+class WC_Admin_Notes_Test_Checkout {
+	/**
+	 * Note traits.
+	 */
+	use NoteTraits;
+
+	/**
+	 * Name of the note for use in the database.
+	 */
+	const NOTE_NAME = 'wc-admin-test-checkout';
+
+	const TASK_LIST_PAYMENTS      = 'woocommerce_task_list_payments';
+	const TASK_LIST_TRACKED_TASKS = 'woocommerce_task_list_tracked_completed_tasks';
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_action( 'update_option_' . self::TASK_LIST_PAYMENTS, array( $this, 'possibly_add_note' ) );
+		add_action( 'update_option_' . self::TASK_LIST_TRACKED_TASKS, array( $this, 'possibly_add_note' ) );
+	}
+
+	/**
+	 * Get the note.
+	 */
+	public static function get_note() {
+		$onboarding_profile = get_option( 'woocommerce_onboarding_profile', array() );
+
+		// Confirm that $onboarding_profile is set.
+		if ( empty( $onboarding_profile ) ) {
+			return;
+		}
+
+		// Make sure that the person who filled out the OBW was not setting up
+		// the store for their customer/client.
+		if (
+			! isset( $onboarding_profile['setup_client'] ) ||
+			$onboarding_profile['setup_client']
+		) {
+			return;
+		}
+
+		// Make sure payments task was completed.
+		$payments_task = get_option( self::TASK_LIST_PAYMENTS, array() );
+		if ( ! isset( $payments_task['completed'] ) ) {
+			return;
+		}
+
+		// Make sure that products were added within the previous 1/2 hour.
+		$query = new \WC_Product_Query(
+			array(
+				'limit'   => 1,
+				'status'  => 'publish',
+				'orderby' => 'date',
+				'order'   => 'ASC',
```

Comment:
We should specify `'return' => 'ids'` here so we aren't querying data unnecessary to get a count.